staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go

```diff
@@ -52,11 +53,38 @@ type C struct {
 	I []interface{} `json:"ci"`
 }
 
-// C needs to implement runtime.Object to make it usable for tests.
+type D struct {
+	A []interface{} `json:"da"`
+}
+
+type E struct {
+	A interface{} `json:"ea"`
+}
+
+type F struct {
+	A string `json:"fa"`
```

Comment:
Add int, float, []string, []int, and []bool, and then be sure to try to send mismatching values to each (int, float, bool -> string, []int -> []string, int -> []string, int -> []int, etc).